### PR TITLE
Promote powerpc64le-unknown-linux-musl to tier 2 with host tools

### PR DESCRIPTION
**Pull request by Gelbpunkt**
MCP: https://github.com/rust-lang/compiler-team/issues/803

I'm using crosstool-ng for building a toolchain because GCC 9 from `musl-toolchain.sh` has float ABI issues (?) and can't compile LLVM, and writing a crosstool-ng config for a target feels less hacky than yet another target specific shell script. I also defined a kernel version, since there wasn't one specified before. If a lower version is desired, just let me know. I also tried to match the rust configure args with the loongarch64 musl tier 2 target.

The resulting compiler works fine, built with `DEPLOY=1 ./src/ci/docker/run.sh dist-powerpc64le-linux` and tested on Alpine Linux in a VM and on a bare metal POWER8 machine:
```
qemu-ppc64le:/tmp/rust-nightly-powerpc64le-unknown-linux-musl$ ash install.sh
install: creating uninstall script at /usr/local/lib/rustlib/uninstall.sh
install: installing component 'rustc'
install: installing component 'rust-std-powerpc64le-unknown-linux-musl'
install: installing component 'cargo'
install: installing component 'rustfmt-preview'
install: installing component 'rls-preview'
install: installing component 'rust-analyzer-preview'
install: installing component 'llvm-tools-preview'
install: installing component 'clippy-preview'
install: installing component 'miri-preview'
install: installing component 'rust-analysis-powerpc64le-unknown-linux-musl'
install: installing component 'llvm-bitcode-linker-preview'
install: WARNING: failed to run ldconfig. this may happen when not installing as root. run with --verbose to see the error

    rust installed.

qemu-ppc64le:~$ echo 'fn main() { println!("hello world"); }' > test.rs
qemu-ppc64le:~$ rustc test.rs
qemu-ppc64le:~$ ./test
hello world
qemu-ppc64le:~$ file test
test: ELF 64-bit LSB executable, 64-bit PowerPC or cisco 7500, OpenPOWER ELF V2 ABI, version 1 (SYSV), statically linked, BuildID[sha1]=596ee6abf9add487ebc54fb71c2076fb6faea013, with debug_info, not stripped
```

try-job: dist-powerpc64le-linux

### CODE REVIEW

File: src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md
Comment (ehuss) [inline src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md:1]: New files need to be added to SUMMARY.md.

**Comment by ehuss** [comment]
r? compiler

File: src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md
Comment (Gelbpunkt) [reply src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md:1]: I added it to src/doc/rustc/src/SUMMARY.md, is there any other one I missed or what exactly do you mean?

Comment (ehuss) [reply src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md:1]: Oh, sorry. I searched and for some reason it didn't highlight in GitHub's UI. Sorry for the noise!

**Comment by Urgau** [comment]
The compiler changes looks go to me, r=me on them.
T-bootstrap for the final review.

r? bootstrap
@bors try

File: src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# riscv64gc-unknown-linux-musl
```
Comment (famfo) [inline src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md:1]:
```suggestion
# powerpc64le-unknown-linux-musl
```

Comment (Gelbpunkt) [reply src/doc/rustc/src/platform-support/powerpc64le-unknown-linux-musl.md]: woopsie

**Review by jieyouxu — APPROVED** [review]
Thanks. The bootstrap bits look fine to me (r=me for bootstrap bits and a second on the compiler changes), I'll hand this off to T-infra to look at the CI changes.

**Comment by jieyouxu** [comment]
r? infra

**Comment by Gelbpunkt** [comment]
The build looks like it got stuck at some point

**Comment by Gelbpunkt** [comment]
...and again

**Comment by Mark-Simulacrum** [comment]
@bors try 

Trying again to try to get a sense of CI time impact...

**Comment by Gelbpunkt** [comment]
The logs are basically missing, and I'm very sure that around ~1h into the job, GitHub stopped updating the output that I was able to see while the job was running (which is now not visible). This is really strange...

**Comment by jieyouxu** [comment]
... Weird.

**Comment by jieyouxu** [comment]
CI no space issue, prob @Mark-Simulacrum or someone on @rust-lang/infra might need to take a look

**Comment by Kobzol** [comment]
We will probably need to use a large runner for this to work. I will bring it up on today's infra meeting.

**Comment by Kobzol** [comment]
Could you please move the CI job to `job-linux-4c-largedisk` in `jobs.yml`? Then we'll evaluate how long it takes.

**Comment by Gelbpunkt** [comment]
> Could you please move the CI job to `job-linux-4c-largedisk` in `jobs.yml`? Then we'll evaluate how long it takes.

Done :)

**Comment by Kobzol** [comment]
It's on the slower side, but ~2h 30m is acceptable given our current CI budget.

**Comment by Gelbpunkt** [comment]
I'm assuming the waiting on author label is left here by mistake.

@rustbot label -S-waiting-on-author +S-waiting-on-review

**Comment by Kobzol** [comment]
No infra objections were raised on [Zulip](https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra/topic/Promote.20.60powerpc64le-unknown-linux-musl.60.20to.20tier.202), so approving.

@bors r=jieyouxu,Urgau,Kobzol

**Pull request merged**